tests/suites/test_suite_x509write.function

```diff
@@ -39,25 +39,29 @@ static int x509_crt_verifycsr( const unsigned char *buf, size_t buflen )
     unsigned char hash[MBEDTLS_MD_MAX_SIZE];
     const mbedtls_md_info_t *md_info;
     mbedtls_x509_csr csr;
+    int ret;
+
+    mbedtls_x509_csr_init( &csr );
 
-    if( mbedtls_x509_csr_parse( &csr, buf, buflen ) != 0 )
-        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+    ret = mbedtls_x509_csr_parse( &csr, buf, buflen );
+    if( ret != 0 )
+        goto exit;
 
     md_info = mbedtls_md_info_from_type( csr.sig_md );
     if( mbedtls_md( md_info, csr.cri.p, csr.cri.len, hash ) != 0 )
     {
         /* Note: this can't happen except after an internal error */
-        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+        ret = MBEDTLS_ERR_X509_BAD_INPUT_DATA;
+        goto exit;
     }
 
-    if( mbedtls_pk_verify_ext( csr.sig_pk, csr.sig_opts, &csr.pk,
-                       csr.sig_md, hash, mbedtls_md_get_size( md_info ),
-                       csr.sig.p, csr.sig.len ) != 0 )
-    {
-        return( MBEDTLS_ERR_X509_CERT_VERIFY_FAILED );
-    }
+    ret = mbedtls_pk_verify_ext( csr.sig_pk, csr.sig_opts, &csr.pk,
+                                 csr.sig_md, hash, mbedtls_md_get_size( md_info ),
+                                 csr.sig.p, csr.sig.len );
```

Comment:
Previously, we'd always return MBEDTLS_ERR_X509_CERT_VERIFY_FAILED no matter what non-zero value we got back from mbedtls_pk_verify_ext(). Now, we can return errors other than MBEDTLS_ERR_X509_CERT_VERIFY_FAILED when mbedtls_pk_verify_ext() can return those. Does this make our tests less robust? Were any of our tests expecting this exact error code and are now more fragile because there may come a time when they don't get that error?

Reply:
`x509_crt_verifycsr` is only used once, in a call `TEST_ASSERT( x509_crt_verifycsr( … ) == 0 )`. So at the moment it doesn't make any difference.

Reply:
OK, I was concerned about the inconsistent approach to keeping or not the errors returned by other functions between the changes around `mbedtls_md()` and `mbedtls_pk_verify_ext()`.